### ISSUE TITLE
Render error messages on /json/... as JSON?

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-core#354]
As HTML/Jinja is the default output render, it is okay that it shows error messages in an HTML-style error message:
![image](https://user-images.githubusercontent.com/16870072/148137547-b7a08fb0-b70f-4ce9-a29c-0c94bbf22fed.png)

But why does a /json/... or /vi/...-call also return the same, HTML-style error message and not some piece of JSON, especially regarding error descriptions which are being lost there.
```json
{
    "code": 404,
    "message": "Not Found",
    "description": "The requested resource could not be found."
}
```
Such a machine-reasable JSON data would be very useful to report back reasons why an action failed back to the user.

Additionally,  this ViUR2-patch would also be reasonable by sending the error message via the response header, and allow Ajax-requests to also deal with error messages in case the returned data should explicitly be HTML.
```patch
--- a/__init__.py
+++ b/__init__.py
@@ -449,6 +449,11 @@ class BrowseHandler(webapp.RequestHandler):
                                raise
                        self.response.clear()
                        self.response.set_status(e.status, e.descr)
+
+                       # Set machine-readable x-viur-error response header in case there is an exception description.
+                       if e.descr:
+                               self.response.headers["x-viur-error"] = e.descr.encode("utf-8") if isinstance(e.descr, unicode) else str(e.descr)
+
                        res = None
                        if conf["viur.errorHandler"]:
                                try:
```

**Comment by phorward** [comment]
> First attempt for a solution:
> 
> https://github.com/viur-framework/viur-core/blob/03e4faff0ed643030d5065485d8a78028798bb91/core/request.py#L327-L331

Hello @ArneGudermann, thanks for your proposal.

Yes, this can work but the URL check on "/json/" is insufficient. I think the routing should generally be done via the render by some kind, but this requires a general rework of the request routing in ViUR.

**Comment by phorward** [comment]
#690 relates as well

**Comment by phorward** [comment]
Fixed by #614
